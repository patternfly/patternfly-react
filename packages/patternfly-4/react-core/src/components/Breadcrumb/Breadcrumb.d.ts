import { FunctionComponent, HTMLProps, ReactNode } from 'react';

export interface BreadcrumbProps extends HTMLProps<HTMLElement> {
    children?: ReactNode;
    className?: string;
    'aria-label'?: string;
}

declare const Breadcrumb: FunctionComponent<BreadcrumbProps>;

export default Breadcrumb;
