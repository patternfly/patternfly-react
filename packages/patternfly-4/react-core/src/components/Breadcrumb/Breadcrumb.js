import React from 'react';
import styles from '@patternfly/patternfly/components/Breadcrumb/breadcrumb.css';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';

const propTypes = {
  /** Children nodes be rendered to the BreadCrumb. Should be of type BreadCrumbItem. */
  children: PropTypes.node,
  /** Additional classes added to the breadcrumb nav. */
  className: PropTypes.string,
  /** Aria-label added to the breadcrumb nav. */
  'aria-label': PropTypes.string,
  /** Additional props are spread to the container <nav> */
  '': PropTypes.any
};

const defaultProps = {
  children: null,
  className: '',
  'aria-label': 'breadcrumb'
};

const Breadcrumb = ({ className, children, ...props }) => (
  <nav {...props} className={css(styles.breadcrumb, className)}>
    <ol className={css(styles.breadcrumbList)}>{children}</ol>
  </nav>
);

Breadcrumb.propTypes = propTypes;
Breadcrumb.defaultProps = defaultProps;

export default Breadcrumb;
