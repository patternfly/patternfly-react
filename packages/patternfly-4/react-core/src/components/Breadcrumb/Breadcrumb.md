---
title: 'Breadcrumb'
cssPrefix: 'pf-c-breadcrumb'
typescript: true
propComponents: ['Breadcrumb', 'BreadcrumbItem', 'BreadcrumbHeading']
---
import { Breadcrumb, BreadcrumbItem, BreadcrumbHeading } from '@patternfly/react-core';

## Simple breadcrumb
```js
import React from 'react';
import { Breadcrumb, BreadcrumbItem, BreadcrumbHeading } from '@patternfly/react-core';

SimpleBreadcrumbs = () => (
  <Breadcrumb>
    <BreadcrumbItem to="#">Section Home</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#" isActive>
      Section Landing
    </BreadcrumbItem>
  </Breadcrumb>
);
```

## Breadcrumb without home link
```js
import React from 'react';
import { Breadcrumb, BreadcrumbItem, BreadcrumbHeading } from '@patternfly/react-core';

WithoutLinkBreadcrumbs = () => (
  <Breadcrumb>
    <BreadcrumbItem>Section Home</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#" isActive>
      Section Landing
    </BreadcrumbItem>
  </Breadcrumb>
);
```

## Breadcrumb with heading
```js
import React from 'react';
import { Breadcrumb, BreadcrumbItem, BreadcrumbHeading } from '@patternfly/react-core';

HeadingBreadcrumbs = () => (
  <Breadcrumb>
    <BreadcrumbItem to="#">Section Home</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbItem to="#">Section Title</BreadcrumbItem>
    <BreadcrumbHeading to="#">Section Title</BreadcrumbHeading>
  </Breadcrumb>
);
```
