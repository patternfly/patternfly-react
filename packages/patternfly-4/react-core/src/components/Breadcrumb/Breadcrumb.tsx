import * as React from 'react';
import styles from '@patternfly/patternfly/components/Breadcrumb/breadcrumb.css';
import { css } from '@patternfly/react-styles';

export interface BreadcrumbProps extends React.HTMLProps<HTMLElement> {
  /** Children nodes be rendered to the BreadCrumb. Should be of type BreadCrumbItem. */
  children?: React.ReactNode;
  /** Additional classes added to the breadcrumb nav. */
  className?: string;
  /** Aria label added to the breadcrumb nav. */
  'aria-label'?: string;
}

export const Breadcrumb: React.FunctionComponent<BreadcrumbProps> = ({
  children = null,
  className = '',
  'aria-label': ariaLabel = 'Breadcrumb',
  ...props
}: BreadcrumbProps) => (
  <nav {...props} aria-label={ariaLabel} className={css(styles.breadcrumb, className)}>
    <ol className={css(styles.breadcrumbList)}>{children}</ol>
  </nav>
);
