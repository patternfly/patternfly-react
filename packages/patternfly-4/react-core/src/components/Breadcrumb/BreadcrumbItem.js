import React from 'react';
import styles from '@patternfly/patternfly-next/components/Breadcrumb/breadcrumb.css';
import PropTypes from 'prop-types';
import { AngleRightIcon } from '@patternfly/react-icons';
import { css, getModifier } from '@patternfly/react-styles';
import { componentShape } from '../../internal/componentShape';

const propTypes = {
  /** Content rendered inside the breadcrumb item. */
  children: PropTypes.node,
  /** Additional classes added to the breadcrumb item. */
  className: PropTypes.string,
  /** HREF for breadcrumb link. */
  to: PropTypes.string,
  /** Flag indicating whether the item is active. */
  isActive: PropTypes.bool,
  /** Target for breadcrumb link. */
  target: PropTypes.string,
  /** Sets the base component to render. Defaults to <a> */
  component: componentShape
};

const defaultProps = {
  children: null,
  className: '',
  to: null,
  isActive: false,
  target: null,
  component: 'a'
};

const BreadcrumbItem = ({ className, children, to, isActive, target, component: Component, ...props }) => (
  <li {...props} className={css(styles.breadcrumbItem, className)}>
    {to && (
      <Component
        href={to}
        target={target}
        className={css(styles.breadcrumbLink, getModifier(styles, isActive && 'current'))}
        aria-current={isActive ? 'page' : undefined}
      >
        {children}
      </Component>
    )}
    {!to && <React.Fragment>{children}</React.Fragment>}
    {!isActive && (
      <span className={css(styles.breadcrumbItemDivider)}>
        <AngleRightIcon />
      </span>
    )}
  </li>
);

BreadcrumbItem.propTypes = propTypes;
BreadcrumbItem.defaultProps = defaultProps;

export default BreadcrumbItem;
