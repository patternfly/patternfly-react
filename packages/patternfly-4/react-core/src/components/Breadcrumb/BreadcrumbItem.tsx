import * as React from 'react';
import { AngleRightIcon } from '@patternfly/react-icons';
import styles from '@patternfly/patternfly/components/Breadcrumb/breadcrumb.css';
import { css, getModifier } from '@patternfly/react-styles';

export interface BreadcrumbItemProps extends React.HTMLProps<HTMLLIElement> {
  /** Content rendered inside the breadcrumb item. */
  children?: React.ReactNode;
  /** Additional classes added to the breadcrumb item. */
  className?: string;
  /** HREF for breadcrumb link. */
  to?: string;
  /** Flag indicating whether the item is active. */
  isActive?: boolean;
  /** Target for breadcrumb link. */
  target?: string;
  /** Sets the base component to render. Defaults to <a> */
  component?: React.ReactNode;
}

export const BreadcrumbItem: React.FunctionComponent<BreadcrumbItemProps> = ({
  children = null,
  className = '',
  to = null,
  isActive = false,
  target = null,
  component = 'a',
  ...props
}) => {
  const Component = component as any;
  return (
    <li {...props} className={css(styles.breadcrumbItem, className)}>
      {to && (
        <Component
          href={to}
          target={target}
          className={css(styles.breadcrumbLink, isActive ? getModifier(styles, 'current') : '')}
          aria-current={isActive ? 'page' : undefined}
        >
          {children}
        </Component>
      )}
      {!to && <React.Fragment>{children}</React.Fragment>}
      {!isActive && (
        <span className={css(styles.breadcrumbItemDivider)}>
          <AngleRightIcon />
        </span>
      )}
    </li>
  );
}