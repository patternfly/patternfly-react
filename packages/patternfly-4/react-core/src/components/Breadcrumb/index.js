export { default as Breadcrumb } from './Breadcrumb';
export { default as BreadcrumbList } from './BreadcrumbList';
export { default as BreadcrumbItem } from './BreadcrumbItem';
export { default as BreadcrumbLink } from './BreadcrumbLink';
export { default as BreadcrumbItemDivider } from './BreadcrumbItemDivider';
export { default as BreadcrumbTitle} from './BreadcrumbTitle';
