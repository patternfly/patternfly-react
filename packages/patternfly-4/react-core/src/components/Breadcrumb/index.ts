export { default as Breadcrumb } from './Breadcrumb';
export { default as BreadcrumbItem } from './BreadcrumbItem';
export { default as BreadcrumbHeading } from './BreadcrumbHeading';
