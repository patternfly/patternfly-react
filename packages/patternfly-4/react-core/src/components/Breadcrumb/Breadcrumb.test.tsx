import * as React from 'react';
import { Breadcrumb } from './Breadcrumb';
import { BreadcrumbItem } from './BreadcrumbItem';
import { shallow } from 'enzyme';

describe('Breadcrumb component', () => {
  test('should render default breadcrumb', () => {
    const view = shallow(<Breadcrumb />);
    expect(view).toMatchSnapshot();
  });

  test('should render breadcrumb with className', () => {
    const view = shallow(<Breadcrumb className="className" />);
    expect(view).toMatchSnapshot();
  });

  test('should render breadcrumb with aria-label', () => {
    const view = shallow(<Breadcrumb aria-label="custom label" />);
    expect(view).toMatchSnapshot();
  });

  test('should render breadcrumb with children', () => {
    const view = shallow(
      <Breadcrumb>
        <BreadcrumbItem to="#">Item 1</BreadcrumbItem> <BreadcrumbItem to="#">Item 1</BreadcrumbItem>
      </Breadcrumb>
    );
    expect(view).toMatchSnapshot();
  });
});
