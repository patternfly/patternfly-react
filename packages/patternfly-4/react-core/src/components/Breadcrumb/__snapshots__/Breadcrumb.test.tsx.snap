// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should render breadcrumb with aria-label 1`] = `
<nav
  aria-label="custom label"
  className="pf-c-breadcrumb"
>
  <ol
    className="pf-c-breadcrumb__list"
  />
</nav>
`;

exports[`Breadcrumb component should render breadcrumb with children 1`] = `
<nav
  aria-label="Breadcrumb"
  className="pf-c-breadcrumb"
>
  <ol
    className="pf-c-breadcrumb__list"
  >
    <BreadcrumbItem
      className=""
      component="a"
      isActive={false}
      target={null}
      to="#"
    >
      Item 1
    </BreadcrumbItem>
     
    <BreadcrumbItem
      className=""
      component="a"
      isActive={false}
      target={null}
      to="#"
    >
      Item 1
    </BreadcrumbItem>
  </ol>
</nav>
`;

exports[`Breadcrumb component should render breadcrumb with className 1`] = `
<nav
  aria-label="Breadcrumb"
  className="pf-c-breadcrumb className"
>
  <ol
    className="pf-c-breadcrumb__list"
  />
</nav>
`;

exports[`Breadcrumb component should render default breadcrumb 1`] = `
<nav
  aria-label="Breadcrumb"
  className="pf-c-breadcrumb"
>
  <ol
    className="pf-c-breadcrumb__list"
  />
</nav>
`;
